src/libs/iowrapper/libuv-wrapper.c

```diff
@@ -0,0 +1,368 @@
+{
+	// fprintf(stderr, "LibUvWrapper_AddTimeout\n");
+	WrapperLibuvData * p = (WrapperLibuvData *)malloc (sizeof (*p));
+	if (p == NULL)
+	{
+		return 1;
+	}
+	memset (p, 0, sizeof (*p));
+	uv_loop_t * loop = (uv_loop_t *)wrapper->private;
+
+	timeoutInfo->ioWrapperData = p;
+	p->timeoutInfo = timeoutInfo;
+	p->wrapper = wrapper;
+	p->timerHandle.data = p;
+
+	int ret = 0;
+	ret = uv_timer_init (loop, &p->timerHandle);
+	if (ret != 0)
+	{
+		return 1;
+	}
+
+	// Start timer if enabled
+	if (timeoutInfo->enabled)
+	{
+		LibUvWrapper_UpdateTimeout (timeoutInfo);
+	}
+
+	return 0;
+}
+
+/**
+ * Remove timeout from IO-Wrapper
+ * @see include/common.h ElektraIoWrapper::removeTimeout
+ */
+static int LibUvWrapper_RemoveTimeout (ElektraIoWrapperTimeoutInfo * timeoutInfo)
+{
+	// fprintf(stderr, "LibUvWrapper_RemoveTimeout\n");
+	WrapperLibuvData * p = (WrapperLibuvData *)timeoutInfo->ioWrapperData;
+	uv_timer_stop (&p->timerHandle);
+	uv_close ((uv_handle_t *)&p->timerHandle, LibUvWrapper_HandleClosedCallback);
+	return 0;
+}
+
+/**
+ * Update idle task in IO-Wrapper
+ * @see include/common.h ElektraIoWrapper::updateIdle
+ */
+static int LibUvWrapper_UpdateIdle (ElektraIoWrapperIdleInfo * idleInfo)
+{
+	// fprintf(stderr, "LibUvWrapper_UpdateIdle (%p, %d)\n", (void*)idleInfo, idleInfo->enabled);
+	WrapperLibuvData * p = (WrapperLibuvData *)idleInfo->ioWrapperData;
+	if (idleInfo->enabled)
+	{
+		uv_idle_start (&p->idleHandle, LibUvWrapper_IdleCallback);
+	}
+	else
+	{
+		uv_idle_stop (&p->idleHandle);
+	}
+	return 0;
+}
+
+/**
+ * Add idle task to IO-Wrapper
+ * @see include/common.h ElektraIoWrapper::addIdle
+ */
+static int LibUvWrapper_AddIdle (ElektraIoWrapper * wrapper, ElektraIoWrapperIdleInfo * idleInfo)
+{
+	// fprintf(stderr, "LibUvWrapper_AddIdle\n");
+	WrapperLibuvData * p = (WrapperLibuvData *)malloc (sizeof (*p));
+	if (p == NULL)
+	{
+		return 1;
+	}
+	memset (p, 0, sizeof (*p));
+	uv_loop_t * loop = (uv_loop_t *)wrapper->private;
+
+	idleInfo->ioWrapperData = p;
+	p->idleInfo = idleInfo;
+	p->wrapper = wrapper;
+	p->timerHandle.data = p;
+
+	int ret = 0;
+	ret = uv_idle_init (loop, &p->idleHandle);
+	if (ret != 0)
+	{
+		return 1;
+	}
+
+	// Add idle to loop if enabled
+	if (idleInfo->enabled)
+	{
+		LibUvWrapper_UpdateIdle (idleInfo);
+	}
+
+	return 0;
+}
+
+/**
+ * Remove idle task from IO-Wrapper
+ * @see include/common.h ElektraIoWrapper::removeIdle
+ */
+static int LibUvWrapper_RemoveIdle (ElektraIoWrapperIdleInfo * idleInfo)
+{
+	// fprintf(stderr, "LibUvWrapper_RemoveIdle\n");
+	WrapperLibuvData * p = (WrapperLibuvData *)idleInfo->ioWrapperData;
+	uv_idle_stop (&p->idleHandle);
+	uv_close ((uv_handle_t *)&p->idleHandle, LibUvWrapper_HandleClosedCallback);
+	return 0;
+}
+
+/**
+ * Cleanup
+ * @param  wrapper [description]
+ * @return         [description]
+ */
+static int LibUvWrapper_Cleanup (ElektraIoWrapper * wrapper)
+{
+	free (wrapper);
+	return 0;
+}
+
+ElektraIoWrapper * LibUvWrapper_Init (uv_loop_t * loop)
```

Comment:
Better `New` instead of `Init`: you are allocating it, not only initializing it.